pkg/cloud/services/ec2/ami.go

```diff
@@ -120,19 +121,29 @@ func (s *Service) defaultAMILookup(amiNameFormat, ownerID, baseOS, kubernetesVer
 		},
 	}
 
-	out, err := s.EC2Client.DescribeImages(describeImageInput)
+	out, err := ec2Client.DescribeImages(describeImageInput)
 	if err != nil {
-		record.Eventf(s.scope.InfraCluster(), "FailedDescribeImages", "Failed to find ami %q: %v", amiName, err)
-		return "", errors.Wrapf(err, "failed to find ami: %q", amiName)
+		return nil, errors.Wrapf(err, "failed to find ami: %q", amiName)
 	}
 	if len(out.Images) == 0 {
-		return "", errors.Errorf("found no AMIs with the name: %q", amiName)
+		return nil, errors.Errorf("found no AMIs with the name: %q", amiName)
 	}
 	latestImage, err := getLatestImage(out.Images)
 	if err != nil {
-		s.scope.Error(err, "failed getting latest image from AMI list")
-		return "", err
+		return nil, err
+	}
+
+	return latestImage, nil
+}
+
+// defaultAMILookup returns the default AMI based on region
+func (s *Service) defaultAMILookup(amiNameFormat, ownerID, baseOS, kubernetesVersion string) (string, error) {
```

Comment:
Is it worth renaming this to something like `defaultAMIIDLookup` to avoid confusion with the new function?